### ISSUE TITLE
[metal] Capture metal frame

### ISSUE THREAD

**Issue by johnson3d** [xamarin/xamarin-macios#6316]
hello,I want to capture GPU frame in an metal app, xcode can capture the app that it run.but ios xamarin is not a xcode project. 
Can i run xamarin app from xcode and capture it?

**Comment by dalexsoto** [comment]
Hello!

What do you mean by capture? Could you show me documentation or a tutorial on the Xcode feature you want?

If you mean taking a video from macOS you can always use QuickTime https://support.apple.com/en-us/HT201066#record to record your devices or if you want a screenshot from the simulator, hitting `cmd + s` keys while on sim should put a screenshot in your `~/Desktop/`. If you need an screenshot from the actual phone, open `Xcode -> Window menu -> Devices and Simulators menu -> Select device on the left panel -> Take Screenshot`.

<img width="258" alt="Screen Shot 2019-06-15 at 7 39 03 AM" src="https://user-images.githubusercontent.com/204671/59551072-0a019480-8f42-11e9-92f4-3264848e54e4.png">

<img width="806" alt="Screen Shot 2019-06-15 at 7 40 58 AM" src="https://user-images.githubusercontent.com/204671/59551075-0c63ee80-8f42-11e9-99c4-729ce2e5c27d.png">

Hope this helps.

**Comment by dalexsoto** [comment]
Ahh never mind, I found the feature you want https://developer.apple.com/documentation/metal/frame_capture_debugging_tools/metal_frame_capture?language=objc 

Unfortunately no, we do not have support it, maybe you could use Xcode instruments' metal tools with your Xamarin app.

I will leave this bug open as a future enhancement that we can research. Thank you for pointing this out.

**Comment by johnson3d** [comment]
Yes! We are working for a cross platform 3D Game engine. So i need Metal Frame Capture to find out my graphics bugs now.
I have tried the Xcod instruments, It's a performance profiler usually. So I am looking forward to get that future enhancement :)
